drivers/usb/udc/Kconfig.mcux

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config USB_UDC_NXP_EHCI
+	bool "NXP MCUX USB EHCI Device controller driver"
+	default y
+	depends on DT_HAS_NXP_EHCI_ENABLED
+	select USB_UDC_NXP_MCUX
+	help
+	  NXP MCUX USB Device Controller Driver for EHCI.
+
+config USB_UDC_NXP_IP3511
```

Comment:
Hi Johann, I want to keep both `USB_UDC_NXP_EHCI` and `USB_UDC_NXP_IP3511` because it is used in other places. For example: the soc.c use it to initialize the usb clock as follow:
```
#if DT_NODE_HAS_STATUS(DT_NODELABEL(usb1), okay) && \
	(CONFIG_USB_DC_NXP_EHCI || CONFIG_USB_UDC_NXP_EHCI)
	CLOCK_EnableUsbhs0PhyPllClock(kCLOCK_Usb480M,
		DT_PROP_BY_PHANDLE(DT_NODELABEL(usb1), clocks, clock_frequency));
	CLOCK_EnableUsbhs0Clock(kCLOCK_Usb480M,
		DT_PROP_BY_PHANDLE(DT_NODELABEL(usb1), clocks, clock_frequency));
#if DT_NODE_HAS_STATUS(DT_NODELABEL(usb1), okay) && CONFIG_USB_DC_NXP_EHCI
	USB_EhciPhyInit(kUSB_ControllerEhci0, CPU_XTAL_CLK_HZ, &usbPhyConfig);
#endif
#endif
```
I have split the udc_mcux.c as udc_mcux_ehci.c and udc_mcux_ip3511.c in order to keep the  both `USB_UDC_NXP_EHCI` and `USB_UDC_NXP_IP3511` . Please share your comments that I should use one udc_mcux.c or use two (udc_mcux_ehci.c and udc_mcux_ip3511.c).